srml/example/src/lib.rs

```diff
@@ -258,13 +258,53 @@ use support::{dispatch::Result, decl_module, decl_storage, decl_event};
 use system::{ensure_signed, ensure_root};
 use codec::{Encode, Decode};
 use sr_primitives::{
-	traits::{SignedExtension, Bounded}, weights::{SimpleDispatchInfo, DispatchInfo},
+	traits::{SignedExtension, Bounded, SimpleArithmetic, SaturatedConversion},
+	weights::{SimpleDispatchInfo, DispatchInfo, DispatchClass, ClassifyDispatch, WeighData, Weight},
 	transaction_validity::{
 		ValidTransaction, TransactionValidityError, InvalidTransaction, TransactionValidity,
 	},
 };
 
-/// Our module's configuration trait. All our types and consts go in here. If the
+/// A custom weight calculator tailored for the dispatch call `set_dummy()` that actually examines
+/// the arguments and makes a decision based on that.
+///
+/// The arguments are passed to `WeighData` as an unbounded generic type `T`. This means that in
+/// order to have any meaningful interpretation, you must explicitly tell the rust compiler what
+/// traits does `T` have. This struct demonstrates how you can customize one _weight calculator_ for
+/// a particular call and fully extract the `T::Balance` argument out of it. This is
+/// quite some boilerplate to determine the weight of just one call. The reasons for that being:
+///   - This is tailored for a call that only accepts a `T::Balance`. If you want to examine the
+///     argument, there must be one struct per call, because these arguments' type are determined at
+///     compile time and it cannot be a _one size fits all_. In other words, the generic type `T` in
+///     `WeighData<T>` can only be one thing at a time. Of course, this depends on how deeply you
+///     want to examine your arguments. As an example, If you just want to interpret your arguments
+///     in an encoded manner, you can bound `T` to `T: Encode` and you will be able to encode `T`
+///     and examine it.
+///   - Note that you this is a non-runtime code and we cannot directly access a `Call` instance,
```

Comment:
```suggestion
///   - Note that you this is a __non-runtime__ code and we cannot directly access a `Call` instance,
```

This is probably nonsense

Reply:
> Note that you this 

that's not english.